### DISCUSSION TITLE
OnUseable events for InteractableObjectBinaryStateEventHook

### DISCUSSION THREAD

**Discussion by Gamingtoday093** [SmartlyDressedGames/Unturned-3.x-Community#4338] [Ideas]
It'd be great if InteractableObjectBinaryStateEventHook (IOBS Event Hook) had events for when it became useable. This would allow objects using IOBS to show if a door is locked for example due to a lack of power or conditions not being met.

Maybe the power part should also be seperated into its own InteractablePowerEventHook

**Comment by DrDallas1** [comment]
That would be client side presumably? (for when it becomes useable?)

**Reply by Gamingtoday093** [reply]
Yes Client sided since it evaluates NPC Conditions

**Comment by SDGNelson** [comment]
My first thought was that it is too performance-intensive to recheck the conditions. Still, on the other hand, object visibility rechecks the conditions a lot, albeit with some considerations for what actually needs to be checked. Maybe that could be improved and shared with other features. No current plan though.
